File: intellij-lsp-dotty/src/DottyFileType.scala
import javax.swing.Icon

import com.intellij.icons.AllIcons
import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.fileTypes.ex.FileTypeIdentifiableByVirtualFile
import com.intellij.openapi.vfs.VirtualFile

object DottyFileType {
  val instance = new DottyFileType
}
class DottyFileType extends LanguageFileType(DottyLanguage.INSTANCE) with FileTypeIdentifiableByVirtualFile {

  override def isMyFileType(file: VirtualFile): Boolean = getDefaultExtension == file.getExtension

  override def getName: String = "Dotty"

  override def getDescription: String = "Dotty files"

  override def getIcon: Icon = AllIcons.Icon_small

  override def getDefaultExtension: String = "scala"
}

File: intellij-lsp-dotty/src/DottyFileType.java
import com.intellij.icons.AllIcons;
import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.fileTypes.ex.FileTypeIdentifiableByVirtualFile;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;
import java.util.Objects;

public class DottyFileType extends LanguageFileType implements FileTypeIdentifiableByVirtualFile {

    public static final DottyFileType INSTANCE = new DottyFileType();

    public DottyFileType() {
        super(DottyLanguage.INSTANCE);
    }

    @Override
    public boolean isMyFileType(@NotNull VirtualFile file) {
        return Objects.equals(getDefaultExtension(), file.getExtension());
    }

    @NotNull
    @Override
    public String getName() {
        return "Dotty";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "Dotty files";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "scala";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return AllIcons.Icon_small;
    }
}
